code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -177,10 +187,52 @@
 						<i>and, most important of all,</i> create new constructs by producing soulstones to capture souls, \
 						and shells to place those soulstones into.</b>"
 
+/mob/living/simple_animal/hostile/construct/builder/Found(atom/A) //what have we found here?
+	if(isliving(A))
+		var/mob/living/L = A
+		if(istype(L, /mob/living/simple_animal/hostile/construct) && L.health < L.maxHealth) //is this a hurt construct? let's go heal it if it is
+			return 1
+		else
+			return 0
+	else
+		return 0
+
+/mob/living/simple_animal/hostile/construct/builder/CanAttack(atom/the_target)
+	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
+		return 0
+	if(Found(the_target) || ..()) //If we Found it or Can_Attack it normally, we Can_Attack it as long as it wasn't invisible
+		return 1 //as a note this shouldn't be added to base hostile mobs because it'll mess up retaliate hostile mobs
+
+/mob/living/simple_animal/hostile/construct/builder/MoveToTarget(var/list/possible_targets)
+	..()
+	if(target && isliving(target))

```

Comment:
see above. you can also remove the check for `target`, as null will fail the istype.